projects/004-exercise-tracker/jimmyhu/src/api/users/usersapi.ts

```diff
@@ -0,0 +1,125 @@
+const exps: any = require('express')
+
+const router = exps.Router()
+
+
+// global variables for store the information
+let users: user[] = []
+let exercises: exercise[] = []
+let invalidID:  number[] = []
+
+
+// user interface, the Id will be a random number for now
+interface user {
+    'username' : string,
+    '_id': number
+}
+
+// exercise extend the user interface due the structure require the same information anyway
+interface exercise extends user {
```

Comment:
Qui vedo uno degli errori che facevo anch'io prima, usare le interfacce per indicare che due oggetti hanno delle proprietà in comune. In realtà le interfacce rappresentano un comportamento che quell'oggetto dovrebbe avere.

Per esempio, su Github ci sono alcuni oggetti come Repositories, Gists, Topics, che estendono l'interfaccia "Starrable", cioè significa che accettano di avere una stellina.
.

Reply:
Ho Provato ad utilizzarli come type ed estenderli come tali, in questo modo è questo che intendeva che avrei dovuto correggere?